cartography/intel/gcp/dns.py

```diff
@@ -0,0 +1,244 @@
+import json
+import logging
+
+from googleapiclient.discovery import HttpError
+
+from cartography.util import run_cleanup_job
+from cartography.util import timeit
+
+logger = logging.getLogger(__name__)
+
+
+@timeit
+def get_dns_zones(dns, project_id):
+    """
+    Returns a list of DNS zones within the given project.
+
+    :type dns: The GCP DNS resource object
+    :param dns: The DNS resource object created by googleapiclient.discovery.build()
+
+    :type project_id: str
+    :param project_id: Current Google Project Id
+
+    :rtype: list
+    :return: List of DNS zones
+    """
+    try:
+        zones = []
+        request = dns.managedZones().list(project=project_id)
+        while request is not None:
+            response = request.execute()
+            for managed_zone in response['managedZones']:
+                zones.append(managed_zone)
+            request = dns.managedZones().list_next(previous_request=request, previous_response=response)
+        return zones
+    except HttpError as e:
+        err = json.loads(e.content.decode('utf-8'))['error']
+        if err.get('status', '') == 'PERMISSION_DENIED':
+            logger.warning(
+                (
+                    "Could not retrieve DNS zones on project %s due to permissions issues. Code: %s, Message: %s"
+                ), project_id, err['code'], err['message'],
+            )
+            return {}
+        else:
+            raise
+
+
+@timeit
+def load_dns_zones(neo4j_session, dns_zones, project_id, gcp_update_tag):
+    """
+    Ingest GCP DNS Zones into Neo4j
+
+    :type neo4j_session: Neo4j session object
+    :param neo4j session: The Neo4j session object
+
+    :type dns_resp: Dict
+    :param dns_resp: A DNS response object from the GKE API
+
+    :type project_id: str
+    :param project_id: Current Google Project Id
+
+    :type gcp_update_tag: timestamp
+    :param gcp_update_tag: The timestamp value to set our new Neo4j nodes with
+
+    :rtype: NoneType
+    :return: Nothing
+    """
+
+    query = """
+    MERGE(zone:GCPDNSZone{id:{ZoneId}})
```

Comment:
Would you be able to refactor this to the `UNWIND` pattern? :-)

Reply:
Done!